src/Strategy/CommonClassesStrategy.php

```diff
@@ -135,4 +136,23 @@ public static function symfonyPsr18Instantiate()
     {
         return new SymfonyPsr18(null, Psr17FactoryDiscovery::findResponseFactory(), Psr17FactoryDiscovery::findStreamFactory());
     }
+
+    /**
+     * Can be used as a condition.
+     * @return bool
+     */
+    public static function isPsr17FactoryInstalled()
+    {
+        try {
+            Psr17FactoryDiscovery::findResponseFactory();
+        } catch (NotFoundException $e) {
+            return false;
+        } catch (\Throwable $e) {
+            trigger_error('Trying to find a PSR-17 ResponseFactory when an exception got thrown: '.$e->getMessage(), E_USER_ERROR);
```

Comment:
Please help me here, should it be `E_USER_ERROR`? Should it be silenced with a `@`?

Reply:
i would make it a E_USER_WARNING `sprintf('Got exception "%s (%s)" while checking if a PSR-17 ResponseFactory is available', get_class($e), $e->getMessage())`

Reply:
if we assume that discovery should try more things in that case. if the psr-17 factory was expected to be viable, then "hiding" the problem in a E_USER_WARNING is not helpful, but if something else is viable and psr-17 is only sort of detected but not viable, the warning makes it so we can continue.

Reply:
> if we assume that discovery should try more things in that case. 

We do.